scintillometry/fourier/base.py

```diff
@@ -0,0 +1,273 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import numpy as np
+import operator
+from astropy.utils import lazyproperty
+
+
+class FFTMakerBase(object):
+    """Base class for all FFT factory classes."""
+
+    _engine_name = None
```

Comment:
Agreed this is ugly. Look at astropys `astropy.coordinates.representation` for another idea (see the `get_name` classmethod - basically strip any `FFTMaker` and lowercase the rest).